cpp/arcticdb/column_store/column.hpp

```diff
@@ -646,6 +646,72 @@ class Column {
         return *res;
     }
 
+    template<typename input_tdt>
+    static void for_each(const Column& input_column,
+                          std::function<void(typename input_tdt::DataTypeTag::raw_type)>&& f) {
+        auto input_data = input_column.data();
+        std::for_each(input_data.cbegin<input_tdt>(), input_data.cend<input_tdt>(), [&f](auto input_value) {
```

Comment:
This is always working on contiguous buffers of memory representing `n` of the same type. Any non-trivial struct would be stored in a Dremel encoding, so this will still hold at the `Column` level.